src/array.cr

```diff
@@ -1549,6 +1549,17 @@ class Array(T)
     dup.sort!
   end
 
+  # Returns an array with all elements in the collection sorted based on the return value of the block given.
+  #
+  #```
+  # a = [3, 1, 2]
+  # b = a.sort { |a,b| b <=> a }
+  #
+  # b # => [3, 2, 1]
+  # a # => [3, 1, 2]
+  # ```
+  # The block given must implement a comparison, either with the comparison operator `<=>`
+  # or a comparison between *a* and *b*, where a < b yields -1, a == b yields 0, and a > b yields 1.
   def sort(&block : T, T -> Int32)
```

Comment:
Could you please add type restrictions to the methods? This is super helpful in the documentation because you can easily see what it returns. (It should also help speed up the compiler's type inference.)